src/App.tsx

```diff
@@ -1,24 +1,215 @@
+  const [addValue, setAddValue] = useState('');
+  const [isInputDisable, setIsInputDisable] = useState(false);
+  const [tempTodo, setTempTodo] = useState<TodoType | null>(null);
+  const [deleteIds, setDeleteIds] = useState<number[]>([]);
+  const [isUpdating, setIsUpdating] = useState(false);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(response => {
+        setTodos(response);
+      })
+      .catch(() => setError('Unable to Load todos'));
+  }, [deleteIds, isUpdating]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(response => {
+        setFilteredTodos(response);
+      })
+      .catch(() => setError('Unable to Load todos'));
+  }, [todos, tempTodo]);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setError('');
+    }, 3000);
+  }, [error]);
+
+  useEffect(() => {
+    switch (filter) {
+      case 'Active': {
+        setFilteredTodos(todos.filter(todo => todo.completed === false));
+        break;
+      }
+
+      case 'Completed': {
+        setFilteredTodos(todos.filter(todo => todo.completed === true));
+        break;
+      }
+
+      default: {
+        setFilteredTodos(todos);
+      }
+    }
+  }, [filter]);
+
+  const handleFilter = (value: string) => {
+    setFilter(value);
+  };
+
+  const handleClose = () => {
+    setError('');
+  };
+
+  const handleAddTodo = (event: FormEvent) => {
+    event.preventDefault();
+
+    if (!addValue) {
+      setError('Title cant be empty');
+    } else {
+      const newTodo: TodoType = {
+        id: 0,
+        userId: USER_ID,
+        title: addValue,
+        completed: false,
+      };
+
+      setTempTodo(newTodo);
+      setIsInputDisable(true);
+
+      addTodo(USER_ID, newTodo).then(() => {
+        setAddValue('');
+        setIsInputDisable(false);
+        setTempTodo(null);
+      }).catch(() => setError('Unable to add a todo'));
+    }
+  };
+
+  const handleRemoveTodo = (id: number) => {
+    removeTodo(id).then(() => {
+      setTodos(todos.filter((todo) => todo.id !== id));
+    }).catch(() => setError('Unable to delete a todo'));
+  };
+
+  const handleComplete = (id: number, completed: boolean) => {
+    updateTodo(id, !completed).then(() => {
+      setIsUpdating(prevState => !prevState);
+    }).catch(() => setError('Unable to update a todo'));
+  };
+
+  const handleAllComplete = () => {
+    const allInCompleted = filteredTodos.filter(todo => (
+      todo.completed === false
+    ));
+
+    if (allInCompleted.length === 0) {
+      filteredTodos.forEach(todo => {
+        updateTodo(todo.id, false).then(() => {
+          setIsUpdating(prevState => !prevState);
+        }).catch(() => setError('Unable to update a todo'));
+      });
+    }
+
+    allInCompleted.forEach(todo => {
+      updateTodo(todo.id, !todo.completed).then(() => {
+        setIsUpdating(prevState => !prevState);
+      }).catch(() => setError('Unable to update a todo'));
+    });
+  };
+
+  const handleEditTitle = (id: number, text: string) => {
```

Comment:
> if new title is the same as the old one just cancel editing;

Reply:
Todo isnt completly removed on the title edit

<img width="606" alt="image" src="https://github.com/mate-academy/react_todo-app-with-api/assets/71132144/be83f90c-2df7-48cd-ade5-6cd4bedb9774">